.github/workflows/ci.yaml

```
@@ -524,7 +524,16 @@ jobs:
             env.UV_CACHE_VERSION }}-${{ steps.generate-uv-key.outputs.version }}-${{
             env.HA_SHORT_VERSION }}-
       - name: Restore apt cache
-        if: steps.cache-venv.outputs.cache-hit != 'true'
+        id: cache-apt-restore
+        uses: actions/cache/restore@v4.2.4
+        with:
+          path: |
```

```suggestion
          lookup-only: true
          path: |
```

As we need only to check if the cache exists
Cache will be downloaded the in the step below

---

The `cache-apt` step is not executed if the cache was hit in the restore step (line 536), this change would break then. The current workflow either just restores the cache with `actions/cache/restore`, or uses the `actions/cache` to populate it in other steps if it doesn't exist.

---

Maybe move the second cache step below the dependency install and change it to `actions/cache/save`?

---

Okay, that's maybe easier to grasp than the implicit behavior of `actions/cache`, also the cache would be created sooner.